File: backend/api/app/schemas/inventory.py
# app/schemas/inventory.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field, field_validator

from ..core.datetime_utils import ensure_utc_aware


class InventoryRecordResponse(BaseModel):
    """库存记录（入库/出库/领用/盘点等）"""

    id: int
    # 事件类型：领用、盘点更新、入库 等
    event_type: str = Field(..., description="出库/入库事件类型：领用、盘点更新等")
    # 入库时间（若有）
    inbound_time: Optional[datetime] = Field(None, description="入库时间")
    # 出库时间（领用时间等）
    outbound_time: Optional[datetime] = Field(None, description="出库时间")
    # 领用人（仅领用事件有值）
    requisitioner_name: Optional[str] = Field(None, description="领用人")
    # 操作人（系统登录用户）
    operator_name: Optional[str] = Field(None, description="操作人")
    # 备件信息
    spare_part_id: int
    location_code: Optional[str] = None
    mes_material_code: Optional[str] = None
    specification_model: Optional[str] = None
    unit: Optional[str] = None
    physical_image_url: Optional[str] = Field(None, description="实物图片1 URL")
    physical_image_url2: Optional[str] = Field(None, description="实物图片2 URL")
    # 数量变化
    quantity: int = Field(..., description="领用/变动数量")
    physical_stock_before: Optional[int] = None
    physical_stock_after: Optional[int] = None
    remark: Optional[str] = None
    requisition_reason: Optional[str] = None
    usage_location: Optional[str] = None
    storage_location: Optional[str] = None

    @field_validator("inbound_time", "outbound_time", mode="before")
    @classmethod
    def _ensure_utc_aware(cls, v):
        return ensure_utc_aware(v) if v is not None else v

    class Config:
        from_attributes = True

File: frontend/src/api/inventory.ts
import request from '@/utils/request'

export interface InventoryRecord {
  id: number
  event_type: string
  inbound_time: string | null
  outbound_time: string | null
  requisitioner_name: string | null
  operator_name: string | null
  spare_part_id: number
  location_code: string | null
  mes_material_code: string | null
  specification_model: string | null
  unit: string | null
  physical_image_url: string | null
  physical_image_url2: string | null
  quantity: number
  physical_stock_before: number | null
  physical_stock_after: number | null
  remark: string | null
  requisition_reason: string | null
  usage_location: string | null
  storage_location: string | null
}

export type TimeRange = 'today' | '7d' | '30d' | '6m' | '1y' | 'custom'

export type MaterialScope = 'electrical' | 'mechanical'

/**
 * 查询库存记录，仅返回指定 scope（电气或机械）数据
 * @param scope 物资范围：electrical 仅电气，mechanical 仅机械
 */
export function getInventoryRecords(
  scope: MaterialScope,
  time_range: TimeRange = '30d',
  start_date?: string,
  end_date?: string,
  event_type?: string,
  requisitioner?: string
) {
  const params: Record<string, string> = { scope, time_range }
  if (start_date) params.start_date = start_date
  if (end_date) params.end_date = end_date
  if (event_type?.trim()) params.event_type = event_type.trim()
  if (requisitioner?.trim()) params.requisitioner = requisitioner.trim()
  return request.get<InventoryRecord[]>('/inventory/records', { params })
}

/** 获取库存记录中已出现的操作人列表，按 scope 仅电气或仅机械 */
export function getInventoryOperatorOptions(scope: MaterialScope) {
  return request.get<string[]>('/inventory/operator-options', { params: { scope } })
}
